Source/WebKit/UIProcess/Automation/libwpe/WebAutomationSessionWPE.cpp

```diff
@@ -0,0 +1,484 @@
+            webModifiers.add(WebEventModifier::AltKey);
+        if (modifiers & WPE_MODIFIER_KEYBOARD_META)
+            webModifiers.add(WebEventModifier::MetaKey);
+        if (modifiers & WPE_MODIFIER_KEYBOARD_CONTROL)
+            webModifiers.add(WebEventModifier::ControlKey);
+        if (modifiers & WPE_MODIFIER_KEYBOARD_SHIFT)
+            webModifiers.add(WebEventModifier::ShiftKey);
+        if (modifiers & WPE_MODIFIER_KEYBOARD_CAPS_LOCK)
+            webModifiers.add(WebEventModifier::CapsLockKey);
+    } else {
+#endif
+        if (modifiers & wpe_input_keyboard_modifier_alt)
+            webModifiers.add(WebEventModifier::AltKey);
+        if (modifiers & wpe_input_keyboard_modifier_meta)
+            webModifiers.add(WebEventModifier::MetaKey);
+        if (modifiers & wpe_input_keyboard_modifier_control)
+            webModifiers.add(WebEventModifier::ControlKey);
+        if (modifiers & wpe_input_keyboard_modifier_shift)
+            webModifiers.add(WebEventModifier::ShiftKey);
+        // libWPE has no Caps Lock modifier.
+#if ENABLE(WPE_PLATFORM)
+    }
+#endif
+
+    return webModifiers;
+}
+
+#if ENABLE(WPE_PLATFORM)
+
+static uint32_t modifiersToEventState(OptionSet<WebEventModifier> modifiers)
+{
+    uint32_t state = 0;
+    if (modifiers.contains(WebEventModifier::ControlKey))
+        state |= WPE_MODIFIER_KEYBOARD_CONTROL;
+    if (modifiers.contains(WebEventModifier::ShiftKey))
+        state |= WPE_MODIFIER_KEYBOARD_SHIFT;
+    if (modifiers.contains(WebEventModifier::AltKey))
+        state |= WPE_MODIFIER_KEYBOARD_ALT;
+    return state;
+}
+
+static unsigned mouseButtonToWPEButton(MouseButton button)
+{
+    switch (button) {
+    case MouseButton::None:
+    case MouseButton::Left:
+        return 1;
+    case MouseButton::Middle:
+        return 3;
+    case MouseButton::Right:
+        return 2;
+    }
+    return 1;
+}
+
+static unsigned stateModifierForWPEButton(unsigned button)
+{
+    uint32_t state = 0;
+
+    switch (button) {
+    case 1:
+        state |= WPE_MODIFIER_POINTER_BUTTON1;
+        break;
+    case 2:
+        state |= WPE_MODIFIER_POINTER_BUTTON2;
+        break;
+    case 3:
+        state |= WPE_MODIFIER_POINTER_BUTTON3;
+        break;
+    case 4:
+        state |= WPE_MODIFIER_POINTER_BUTTON4;
+        break;
+    case 5:
+        state |= WPE_MODIFIER_POINTER_BUTTON5;
+        break;
+    default:
+        break;
+    }
+
+    return state;
+}
+#endif // ENABLE(WPE_PLATFORM)
+
+static WebCore::IntPoint deviceScaleLocationInView(WebPageProxy& page, const WebCore::IntPoint& locationInView)
+{
+    WebCore::IntPoint deviceScaleLocationInView(locationInView);
+    deviceScaleLocationInView.scale(page.deviceScaleFactor());
+    return deviceScaleLocationInView;
+}
+
+#if ENABLE(WEBDRIVER_MOUSE_INTERACTIONS)
+
+#if ENABLE(WPE_PLATFORM)
+static void doMouseEvent(WebPageProxy& page, const WebCore::IntPoint& location, unsigned button, unsigned state, uint32_t modifiers)
```

Comment:
This is new api specific, instead of unsigned state we could receive an event type, or if we want to be more specific to mouse events, it could be bool isPressed or something like that

Reply:
Went with `bool isPressed`.

Reply:
I still see unsigned state, here